File: tests/XRoadLib.Tests/SoapFaultCodesTest.fs
namespace XRoadLib.Tests

open FsUnit
open NUnit.Framework
open XRoadLib.Soap

[<TestFixture>]
module SoapFaultCodesTest =
    [<Test>]
    let ``client fault has correct code`` () =
        let faultCode = ClientFaultCode("message content")
        faultCode.Value |> should equal "Client.message content"

    [<Test>]
    let ``server fault has correct code`` () =
        let faultCode = ServerFaultCode("server message content")
        faultCode.Value |> should equal "Server.server message content"

    [<Test>]
    let ``without message code`` () =
        let faultCode = ClientFaultCode()
        faultCode.Value |> should equal "Client"

    [<Test>]
    let ``internal server error`` () =
        let faultCode = ServerFaultCode.InternalError
        faultCode.Value |> should equal "Server.InternalError"

File: test/XRoadLib.Tests/Soap/SoapFaultCodesTest.cs
using XRoadLib.Soap;

namespace XRoadLib.Tests.Soap;

public class SoapFaultCodesTest
{
    [Fact]
    public void ClientFaultHasCorrectCode()
    {
        var faultCode = new ClientFaultCode("message content");
        Assert.Equal("Client.message content", faultCode.Value);
    }

    [Fact]
    public void ServerFaultHasCorrectCode()
    {
        var faultCode = new ServerFaultCode("server message content");
        Assert.Equal("Server.server message content", faultCode.Value);
    }

    [Fact]
    public void WithoutMessageCode()
    {
        var faultCode = new ClientFaultCode();
        Assert.Equal("Client", faultCode.Value);
    }

    [Fact]
    public void InternalServerError()
    {
        var faultCode = ServerFaultCode.InternalError;
        Assert.Equal("Server.InternalError", faultCode.Value);
    }
}